src/common/src/types/interval.rs

```diff
@@ -1001,6 +1000,24 @@ impl ToText for crate::types::Interval {
     }
 }
 
+/// Error type for parsing an [`Interval`].
+#[derive(thiserror::Error, Debug, thiserror_ext::Construct)]
```

Comment:
Is there any big difference between `IntervalParseError::Invalid` and `IntervalParseError::invalid`? Given that the former one is also a special function?

Reply:
One can pass any `impl Into<String>` with constructors. Also, if there's a `backtrace` field, it can also be automatically generated.

```rust
#[automatically_derived]
impl IntervalParseError {
    /// Constructs a [`IntervalParseError::Invalid`] variant.
    pub fn invalid(arg_0: impl Into<String>) -> Self {
        IntervalParseError::Invalid { 0: arg_0.into() }.into()
    }
}
```